### ISSUE TITLE
Request: add secondary context for "Not Interested" to help user build recommendation via YouTube algo

### ISSUE THREAD

**Issue by Jahfry** [yuliskov/SmartTubeNext#825]
**Request:** additonal context for "Not Interested"

The 2 options the standard web/mobile client provides are:

*  "I've already watched this video"
*  "I don't like this video"

***Note:*** rather than adding these as a secondary press(actually tertiary given the long-press), these could be added as additional entries in the initial context menu.

*Reason:* In my case most times I'd be using "already watched" after clearing a few years of viewing data and seeing many videos suggested that I've already seen. That setting helps tell the YouTube algorithm "I've already seen it but yes, I like this".

The default "Not Interested" without selecting a reason trains the algorithm more to think "they don't like this" so using Smart Tube Next's "Not Interested" too often eventually starts filtering out topics similar to the secondary context of "I don't like this video" (seems to take longer than when specifically selecting that, but it does happen over time). 

*Related/additional detail*: another option in the context menu that would be good would be 

* Don't recommend this channel

 I'm not sure I've ever even found a way in Smart to do this and YouTube has some really bad channel recommendations randomly appear at times. 

Adding 3 more options does start to bloat the context menu significantly, and I could see that being a problem. I'm addressing a possible way to get around that in this suggestion:

Incorporating that option to pick and choose which options we can see in the menu would allow the user to keep the clutter down. If you want to not enable these by default, then the default UI can remain as-is,  with proposed items as:

*Note:* [X] means enable by default, [ ] means available for user to enable, ↕ is the button for reordering actions. 

* [X] ↕ Remove, Not interested
* [ ] ↕ Remove, already watched
* [ ] ↕ Remove, dislike
* [ ] ↕ Remove, don't recommend Channel

Another option, if there's a problem with allowing users to disable/enable context menu items (see https://github.com/yuliskov/SmartTubeNext/issues/824 ) , would be to leave "Not Interested" as-is in the menu and when selected have it trigger an additional context menu of the other 3 options. This is definitely not preferred as it adds another step to usage, but I'm not familiar with your data structure to know if the preferred way is significantly harder. 

These 2 requests were added as pieces that may be useful when thinking of how to implement this request (which is by far my most desired function):

* https://github.com/yuliskov/SmartTubeNext/issues/823
* https://github.com/yuliskov/SmartTubeNext/issues/824

**Comment by Jahfry** [comment]
Revisiting this. I was relying on "Not Interested" for the months since this request and it has really messed up my recommendations. 

Really really would like to see "I've already watched this video" added to the long press menu. Otherwise the suggestions list is mostly full of videos that I've already watched either in part or in full.

**Comment by Jahfry** [comment]
Alternative request that would help with this: #1539

**Comment by costinmx** [comment]
Unfortunately, Yutube has missed the most used real answer: "I'm really really JUST not intrested in this video"
"I did not watched it, so I and cannot dislike it or unrecommend it. JUST not interested."
Forcing you to choose between two non-true options, guarantees for a fake answer and this does in fact harm the algortyhm.

**Comment by Jahfry** [comment]
True. But having a choice between the two evils is better than always using the more evil (to the Algo) "not interested" on topics that you'd like to still show interest in (and I often have already watched a video being suggested, either on a different account or prior to the watch history auto deleting after a long while).
